File: src/main/programs/strings/CountVowels.java
package strings;

import java.util.Scanner;

public class CountVowels {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a string: ");
        String input = scanner.nextLine();
        countVowelsAndConsonants(input);
        scanner.close();
    }

    public static void countVowelsAndConsonants(String str) {
        int vowels = 0;
        int consonants = 0;

        // Convert string to lowercase
        str = str.toLowerCase();

        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);

            // Check if character is a vowel
            if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') {
                vowels++;
            }
            // Check if character is a consonant
            else if (Character.isLetter(ch)) {
                consonants++;
            }
        }

        System.out.println("Input String: " + str);
        System.out.println("Number of vowels: " + vowels);
        System.out.println("Number of consonants: " + consonants);
    }
}

File: src/main/programs/strings/CountVowels.kt
package strings

fun countVowelsAndConstants(string: String){
    var clnStr=string.lowercase().filter { it.isLetter() }
    val vowels=clnStr.count { it in "aeiou" }
    val const=clnStr.count{it.isLetter() && it !in "aeiou"}
    println("Input String: $string")
    println("Vowels: $vowels")
    println("Constants: $const")


}

fun countVowelsAndConsonants(str: String) {
    var vowels = 0
    var consonants = 0

    val text = str.lowercase()

    for (char in text) {
        when {
            char in "aeiou" -> vowels++
            char.isLetter() -> consonants++
        }
    }

    println("Input String: $str")
    println("Number of vowels: $vowels")
    println("Number of consonants: $consonants")
}

fun main(){
    val testStrings = listOf(
        "Hello World!",
        "OpenAI",
        "12345",
        "aEiOu",
        ""
    )
    countVowelsAndConstants(testStrings.toString())
    countVowelsAndConsonants(testStrings.toString())
}